File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples18.vb
' <snippetlinqexamples18>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim methodResults = svcContext.ContactSet _
                      .Where(Function(a) a.LastName.Equals("Smith"))
  Dim methodResults2 = svcContext.ContactSet _
                       .Where(Function(a) a.LastName.StartsWith("Smi"))
  Console.WriteLine()
  Console.WriteLine("Method query using Lambda expression")
  Console.WriteLine("---------------------------------------")
  For Each a In methodResults
   Console.WriteLine("Name: " &amp; a.FirstName &amp; " " &amp; a.LastName)
  Next a
  Console.WriteLine("---------------------------------------")
  Console.WriteLine("Method query 2 using Lambda expression")
  Console.WriteLine("---------------------------------------")
  For Each a In methodResults2
   Console.WriteLine("Name: " &amp; a.Attributes("firstname").ToString() _
                     &amp; " " &amp; a.Attributes("lastname").ToString())
  Next a
 End Using

' </snippetlinqexamples18>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples18.cs
// <snippetlinqexamples18>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var methodResults = svcContext.ContactSet
   .Where(a => a.LastName == "Smith");
  var methodResults2 = svcContext.ContactSet
   .Where(a => a.LastName.StartsWith("Smi"));
  Console.WriteLine();
  Console.WriteLine("Method query using Lambda expression");
  Console.WriteLine("---------------------------------------");
  foreach (var a in methodResults)
  {
   Console.WriteLine("Name: " + a.FirstName + " " + a.LastName);
  }
  Console.WriteLine("---------------------------------------");
  Console.WriteLine("Method query 2 using Lambda expression");
  Console.WriteLine("---------------------------------------");
  foreach (var a in methodResults2)
  {
   Console.WriteLine("Name: " + a.Attributes["firstname"] +
    " " + a.Attributes["lastname"]);
  }
 }

// </snippetlinqexamples18>